python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -750,29 +622,14 @@ def fit(self, X, y, sample_weight=None, bags=None, init_score=None):  # noqa: C9
 
         # branch so we have no correlation to the binning rng that uses the same seed
         rng = native.branch_rng(native.create_rng(seed))
-        used_seeds = set()
-        rngs = []
-        internal_bags = []
-        for idx in range(self.outer_bags):
-            while True:
-                bagged_rng = native.branch_rng(rng)
-                check_seed = native.generate_seed(bagged_rng)
-                # We do not want identical bags. branch_rng is pretty good at avoiding
-                # collisions, but it is not a cryptographic RNG, so it is possible.
-                # Check with a 32-bit seed if we have a collision and regenerate if so.
-                if check_seed not in used_seeds:
-                    break
-            used_seeds.add(check_seed)
-
-            if bags is None:
-                bag = make_bag(
-                    y,
-                    self.validation_size,
-                    bagged_rng,
-                    0 <= n_classes and not is_differential_privacy,
-                )
-            else:
-                bag = bags[idx]
+        rngs = _create_rngs(rng, n_rngs=self.outer_bags)
+        if bags is None:
+            stratified = n_classes >= 0
```

Comment:
stratified = n_classes >= 0

should instead be:

stratified = n_classes >= 0 and not is_differential_privacy